html/apps/barchart/services/analyticService.js

```diff
@@ -0,0 +1,202 @@
+angular.module('myApp')
+.factory('analyticService', 
+    function($localStorage, $moment){
+        // returns an object of {word:freq} pairs
+        
+        return {
+            getNewTweets: getNewTweets,
+            updateWordFreq: updateWordFreq,
+            updateMentionFreq: updateMentionFreq,
+            updateHashtagDateFreq: updateHashtagDateFreq
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["updateHashtagDateFreq" was used before it was defined](https://www.codacy.com/app/sudheesh1995/loklak_server/file/2626095151/issues/source?bid=3288751&fileBranchId=3313532#l10)